### DISCUSSION TITLE
state reset on Model::loadCount() in v1 branch

### DISCUSSION THREAD

**Discussion by aleksandrpnshkn** [spatie/laravel-model-states#179] [Bugs]
```sh
laravel new states-bug
cd states-bug
composer require spatie/laravel-model-states "1.9.1"
```
Add column and create relations table in `database/migrations/2014_10_12_000000_create_users_table.php`:
```php 
public function up() {
    Schema::create('users', function (Blueprint $table) {
        // ... default columns
        $table->string('state');
    });
    
    Schema::create('relations', function (Blueprint $table) {
        $table->unsignedBigInteger('user_id');
    });
}
```
Create states:
```php
use Spatie\ModelStates\State;

abstract class UserState extends State {}
```
```php
class DefaultState extends UserState {}
```
```php
class AnotherState extends UserState {}
```
Register state in `\App\Models\User`:
```php
use HasStates;

protected function registerStates(): void {
    $this->addState('state', UserState::class)
        ->allowTransition(DefaultState::class, AnotherState::class)
        ->default(DefaultState::class);
}
```
Make Relation model:
```shell
php artisan make:model Relation
```
Add relation to User: 
```php
public function relations() {
    return $this->hasMany(Relation::class);
}
```
Uncomment sqlite in `phpunit.xml`:
```
<server name="DB_CONNECTION" value="sqlite"/>
<server name="DB_DATABASE" value=":memory:"/> 
```
Create and run feature test:
```php
use RefreshDatabase;

public function test_retrieved_with_correct_state() {
    $userWithNonDefaultState = User::factory()->create([
        'state' => AnotherState::class,
    ]);

    $this->assertFalse($userWithNonDefaultState->state->equals(DefaultState::class), 'Passed');

    $userWithNonDefaultState->loadCount('relations');

    $this->assertFalse($userWithNonDefaultState->state->equals(DefaultState::class), 'Failed');
}
```
Fail.

In the `\Spatie\ModelStates\HasStates::bootHasStates()` method there is callback `$unserializeState`. On `Model::loadCount(...)` the `retrieved` event will be fired only with attributes `id` and `relations` in the model. There is no attribute `state`, but the callback tries to resolve state class for it. `$model->getAttribute('state')` returns null, so `::resolveStateClass(...)` returns the default state.
I added this code to the first line of the `$unserializeState` callback:
```php
if (! array_key_exists($stateConfig->field, $model->getAttributes())) {
    return;
}
```
Now the test passes. The tests from the library are also successful.
Can this fix break anything else? Can it be commited to the repository?

**Comment by brendt** [comment]
We're not supporting v1 anymore, especially since it did some very wonky things as a replacement for custom casts in Laravel 8.
